frontend/src/components/RecipeDraft.js

```diff
@@ -1,16 +1,52 @@
-import {Button, makeStyles, TextField} from "@material-ui/core";
+import {Button, IconButton, makeStyles, TextField} from "@material-ui/core";
 import {useState} from "react";
 import IngredientEditor from "./IngredientEditor";
+import AddCircleOutlineOutlinedIcon from '@material-ui/icons/AddCircleOutlineOutlined';
+import {CheckOutlined} from "@material-ui/icons";
+import styled from "styled-components/macro";
 
-export default function RecipeDraft({recipe, onRecipeNameUpdated, onIngredientsUpdated,  saveDraft}) {
+export default function RecipeDraft({recipe, onRecipeNameUpdated, onIngredientsUpdate, saveDraft}) {
     const [ingredients, setIngredients] = useState(recipe.ingredients);
 
     const classes = useStyles();
 
     const onIngredientUpdated = (updatedIngredient) => {
         setIngredients(ingredients.map(ingredient =>
             ingredient.id === updatedIngredient.id ? updatedIngredient : ingredient))
-        onIngredientsUpdated(ingredients)
+        onIngredientsUpdate(ingredients.map(ingredient =>
+            ingredient.id === updatedIngredient.id ? updatedIngredient : ingredient))
+    }
+
+    const onAddIngredient = () => {
+        const id = uuid();
+        setIngredients(
```

Comment:
Where are these values actually set? Because here it looks like you just put empty strings ...

Reply:
this adds a new extra ingredient to a recipe, the other values are filled in in another component by userinput.